client/src/main/java/org/spine3/base/Identifier.java

```diff
@@ -0,0 +1,322 @@
+            }
+
+            @Override
+            boolean matchMessage(Message message) {
+                return message instanceof StringValue;
+            }
+
+            @Override
+            <I> boolean matchClass(Class<I> idClass) {
+                return String.class.equals(idClass);
+            }
+
+            @Override
+            <I> Message toMessage(I id) {
+                return newStringValue((String)id);
+            }
+
+            @Override
+            String fromMessage(Message message) {
+                return ((StringValue)message).getValue();
+            }
+
+            @Override
+            <I> I getDefaultValue(Class<I> idClass) {
+                return (I) "";
+            }
+        },
+
+        INTEGER {
+            @Override
+            <I> boolean matchValue(I id) {
+                return id instanceof Integer;
+            }
+
+            @Override
+            boolean matchMessage(Message message) {
+                return message instanceof UInt32Value;
+            }
+
+            @Override
+            <I> boolean matchClass(Class<I> idClass) {
+                return Integer.class.equals(idClass);
+            }
+
+            @Override
+            <I> Message toMessage(I id) {
+                return UInt32Value.newBuilder()
+                                  .setValue((Integer) id)
+                                  .build();
+            }
+
+            @Override
+            Integer fromMessage(Message message) {
+                return ((UInt32Value)message).getValue();
+            }
+
+            @Override
+            <I> I getDefaultValue(Class<I> idClass) {
+                return (I) Integer.valueOf(0);
+            }
+        },
+
+        LONG {
+            @Override
+            <I> boolean matchValue(I id) {
+                return id instanceof Long;
+            }
+
+            @Override
+            boolean matchMessage(Message message) {
+                return message instanceof UInt64Value;
+            }
+
+            @Override
+            <I> boolean matchClass(Class<I> idClass) {
+                return Long.class.equals(idClass);
+            }
+
+            @Override
+            <I> Message toMessage(I id) {
+                return UInt64Value.newBuilder()
+                                  .setValue((Long) id)
+                                  .build();
+            }
+
+            @Override
+            Long fromMessage(Message message) {
+                return ((UInt64Value)message).getValue();
+            }
+
+            @Override
+            <I> I getDefaultValue(Class<I> idClass) {
+                return (I) Long.valueOf(0);
+            }
+        },
+
+        MESSAGE {
+            @Override
+            <I> boolean matchValue(I id) {
+                return id instanceof Message;
+            }
+
+            @Override
+            boolean matchMessage(Message message) {
+                return !(message instanceof StringValue
```

Comment:
This expression worth commenting. 

In particular, it's not clear that this list should be appended with another expression in case the `Type` enum gets another member.